web/server.go

```diff
@@ -30,24 +32,31 @@ func splitHost(host string) (instanceHost string, appSlug string) {
 // Create returns a new web server that will handle that apps routing given the
 // host of the request.
 func Create(router *echo.Echo, serveApps echo.HandlerFunc) (*echo.Echo, error) {
-	appsHandler := middlewares.LoadSession(serveApps)
+	appsHandler := middlewares.Compose(serveApps,
+		middlewares.Secure(&middlewares.SecureConfig{
+			HSTSMaxAge:    365 * 24 * time.Hour, // 1 year
+			CSPScriptSrc:  []middlewares.CSPSource{middlewares.CSPSrcSelf, middlewares.CSPSrcParent},
+			CSPConnectSrc: []middlewares.CSPSource{middlewares.CSPSrcSelf, middlewares.CSPSrcParent},
+			CSPFrameSrc:   []middlewares.CSPSource{middlewares.CSPSrcParent},
+			XFrameOptions: middlewares.XFrameDeny,
```

Comment:
Just to be sure, XFrameDeny is temporary. When we will have intents, we will allow the apps/routes with an intent to be embedded in an iframe for inter-apps. Right?